test/functional/interface_zmq.py

```diff
@@ -204,5 +205,30 @@ def test_reorg(self):
         # And the current tip
         assert_equal(hashtx.receive().hex(), self.nodes[1].getblock(connect_blocks[0])["tx"][0])
 
+    def test_multiple_interfaces(self):
+        import zmq
+
+        # Set up two subscribers with different addresses
+        subscribers = []
+        for x in range(2):
+            address = 'tcp://127.0.0.1:%d' % (28334 + x)
+            socket = self.ctx.socket(zmq.SUB)
+            socket.set(zmq.RCVTIMEO, 60000)
+            hashblock = ZMQSubscriber(socket, b"hashblock")
+            socket.connect(address)
+            subscribers.append({'address': address, 'socket': socket, 'hashblock': hashblock})
+
+        self.restart_node(0, ['-zmqpub%s=%s' % (subscribers[0]['hashblock'].topic.decode(), subscribers[0]['address']),
```

Comment:
maybe compress this into:

`self.restart_node(0, ['-zmqpub%s=%s' % (subscribers[i]['hashblock'].topic.decode(), subscribers[i]['address']) for i in range(2)])`

or there-about(didn't check parenthesis lining up :) )

Reply:
Thanks, applied your change slightly modified along with some other minor changes :)